src/components/TutorialView.jsx

```diff
@@ -96,10 +89,8 @@ class TutorialView extends React.Component {
   }
 
   isFinalStep() {
-    const currentStep = (this.stepThrough && this.stepThrough.state && this.stepThrough.state.step)
-      ? this.stepThrough.state.step + 1 : 0;
-    const maxSteps = (this.stepThrough && this.stepThrough.props && this.stepThrough.props.children)
-      ? React.Children.count(this.stepThrough.props.children) : 0;
+    const currentStep = this.state.stepIndex;
+    const maxSteps = this.props.tutorial.steps.length - 1 || 0;
```

Comment:
I'm not sure what the point of this logic is; if `this.props.tutorial.steps.length` is 0, then `maxSteps` will be `-1`, because in JS, `-1 || 0` will result in `-1`. Did you mean `Math.max(this.props.tutorial.steps.length, 0)` ?

Reply:
Ah wait, I get what's happening here - you're worried that `this.props.tutoria.steps.length` might be undefined. If that's the case, we should check for the existence of `this.props.tutoria.steps`, since `.length` should be a given if `steps` is an Array

Reply:
This logic needs to be cleaned up generally. Yes, that `||` was mainly being used as an existential check.